### ISSUE TITLE
Bitwise & / | / ^ of a bool operand with an int emits CS0019 (bool op int) at Full

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1452]
## Goal

Over-raise correctness bug for the #1356 burndown. A bitwise `and`/`or`/`xor` IL op that combines a **`bool`-typed operand** (a `Comparison` result / a bool-typed slot) with an **`int`-typed operand** is raised to a C# `&`/`|`/`^` whose operands print as `bool` and `int`. C# has no such operator → CS0019. The pipeline reports `Full`.

## The over-raise claim being narrowed

This is the inverse of the #1403 family: the *result* type is patched but the *operand spelling* is not. On the IL stack both operands are `i4` (0/1), so `and`/`or`/`xor` is legal. `TypeFamilies.BinaryResult` (`TypeFamilies.cs:51-70`) detects the bool/non-bool mix and returns the non-bool operand's type so the binary's *result* reads as `int` — its own comment even anticipates the `(a > 0) | (b ? 2 : 0)` shape. But the bool operand keeps its faithful `System.Boolean` type (`Comparison.ResultType` => Boolean, `IrNodes.cs:802/827`) and prints bare. The printer's `BinaryBody` (`CSharpPrinter.Numerics.cs:64-104`) coerces operands for the enum/integer mix (lines 71-77) and the signed/unsigned same-width mix (lines 89/115), but has **no case for the bool/integer mix**, so the bool operand is emitted as a bare `bool` beside an `int`.

## Minimal counterexample (standard csc, not synthetic)

```csharp
public static int AndMix(int a, bool b)    => (a > 0 ? 1 : 0) & (b ? 1 : 0);
public static int BoolOrInt(int a, bool b) => (a > 0 ? 1 : 0) | (b ? 2 : 0);
```

Driving the real pipeline produces, at `Fidelity=Full`:

| source | decompiler emits |
| --- | --- |
| `AndMix` | `return (a > 0) & (b ? 1 : 0);` |
| `BoolOrInt` | `bool S_0 = a > 0; int S_1 = b ? 2 : 0; return S_0 \| S_1;` |

Recompiling the output fails:

```
error CS0019: Operator '&' cannot be applied to operands of type 'bool' and 'int'
error CS0019: Operator '|' cannot be applied to operands of type 'bool' and 'int'
```

(The `S_0 | S_1` form additionally surfaces an unassigned-slot artifact in some shapes — same root: a bool-typed slot landing in an integer bitwise context.) Invalid C# at `Full`.

## Missing gate

`BinaryBody` does not materialize a `bool` operand of an integer bitwise `&`/`|`/`^` into an integer expression (`operand ? 1 : 0`); the bool/integer mix is the one operand-coercion case it doesn't handle.

## First fix shape

In `CSharpPrinter.Numerics.cs` `BinaryBody`, when a bitwise `&`/`|`/`^` has exactly one `Boolean` operand and one integer operand, materialize the bool operand as `(operand ? 1 : 0)` (mirroring the existing enum-cast and unsigned-reinterpret coercions). If that's deemed out of policy, lower fidelity instead. The all-bool case (`bool & bool`) must stay a valid logical/bitwise bool op and is unaffected.

## Done signal

- Adversarial fixtures: `(a > 0 ? 1 : 0) & (b ? 1 : 0)` and `(a > 0 ? 1 : 0) | (b ? 2 : 0)` recompile (bool operand materialized to `(… ? 1 : 0)`).
- Positive canaries: `bool & bool` / `bool | bool` still print as a plain bool op; `int & int` unchanged.

## Notes

Confirmed end-to-end through the product pipeline and round-tripped through csc (CS0019). Trigger is ordinary C#. Type-inference/printer-coercion class, the operand-side complement of #1403 and #1425. Root inference is correct (`BinaryResult` deliberately sinks the result to `int`); the gap is purely the missing operand materialization in the printer.
